flytekit/core/context_manager.py

```diff
@@ -359,11 +360,19 @@ def __getattr__(self, item: str) -> _GroupSecrets:
 
     def get(
         self,
+        *,
```

Comment:
Adding a `*` right away is backward breaking. This means that old code using position arguments like `secret.get("my_group", "key")` will stop working. 

I implemented a similar deprecation logic in scikit-learn here: https://github.com/scikit-learn/scikit-learn/blob/44017bb4a09359c22d4810c870fd69fbd5673e63/sklearn/utils/validation.py#L33 that you copy into flytekit.

The decorator can be used like this, and it'll raise the correct warning based on where the `*` is, but also continue to support positional arguments.

```python
@_deprecate_positional_args(version="...")
def get(self, *, ...)
```

Reply:
Hi @thomasjpfan, Thank you for the feedback! I’ll implement the decorator for handling deprecation warnings, similar to the logic used in scikit-learn, to maintain backward compatibility. I’ll update the PR shortly with these changes.

Reply:
Hi @thomasjpfan, I’ve implemented the decorator based on the structure and logic from the scikit-learn reference to ensure backward compatibility, issuing a FutureWarning for positional arguments as discussed. While the core logic seems aligned, it didn’t fully pass tests, even with debug statements added for troubleshooting.

Any feedback or guidance on fine-tuning or spotting potential issues with the test outcomes would be a great help!

Regards

Reply:
cc @eapolinario